rpe/resources/gcp.py

```diff
@@ -104,6 +101,8 @@ def _extract_cai_name_data(name):
             "cluster": r"/clusters/([^\/]+)/",
             # ServiceAccounts
             "service_account": r"serviceAccounts/([^\/]+)/",
+            # Dataform
+            "repository": r"/repositories/([^\/]+)/",
```

Comment:
this seems to handle both workspace and repository, should we split them into their own resource data?

Reply:
Here's a sample dataform repository cai name
`name= "//dataform.googleapis.com/projects/test-project/locations/us-central1/repositories/test-repository"`
here's what the resource_data looks like after executing the loop
`{'project_id': 'test-project', 'location': 'us-central1', 'name': 'test-repository'}`

sample dataform workspace cai name
name= "//dataform.googleapis.com/projects/test-project/locations/us-central1/repositories/test-repository/workspaces/test-workspace"
resource_data after executing loop
`{'project_id': 'test-project', 'location': 'us-central1', 'name': 'test-workspace', 'repository': 'test-repository'}`

Reply:
We need the repository field to query the dataform api for workspaces

Reply:
the code gives the impression that it will set the repository in both repository and workspace and that doesnt seem to be the case, can you document this behaviour..

Reply:
added a comment..